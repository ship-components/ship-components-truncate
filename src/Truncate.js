import React from 'react';
import PropTypes from 'prop-types';
import truncateString from './truncateString';

export default function Truncate(props) {
  const {
    className,
    onClick,
    text,
    characterLimit
  } = props;

  return (
    <span
      className={className}
      onClick={onClick}
    >
      {truncateString(text, characterLimit)}
    </span>
  );
}

/**
 * Default
 * @type    {Object}
 */
Truncate.defaultProps = {
  className: undefined,
  onClick: undefined
};

/**
 * Types
 * @type    {Object}
 */
Truncate.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func,
  text: PropTypes.string.isRequired,
  characterLimit: PropTypes.number.isRequired
};
