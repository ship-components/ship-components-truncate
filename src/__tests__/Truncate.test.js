// README FIRST:
// 1. Uncomment the codes below to use it as an example
// 2. You can change "Index" on line 10 to anything you wish to be called
// 3. Please note that a lot of the tests below are just for reference - may or maynot be relevent to your react app. Feel free to remove them.
// 4. Examples below are using Jest with enzyme but feel free to use anything you wish.

import React from 'react';
import Truncate from '../Truncate'
import renderer from 'react-test-renderer'

  describe('Truncate', () => {

    it('should be a function', () => {
      expect(typeof Truncate).toBe('function')
    });

    it('renders correctly', () => {
      const tree = renderer
        .create(<Truncate text="lorem ipsum something that i dont remember" characterLimit={20} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });




  });
  
