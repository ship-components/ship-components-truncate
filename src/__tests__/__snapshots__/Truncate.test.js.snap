// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Truncate renders correctly 1`] = `
<span>
  lorem ipsum...
</span>
`;
